README.md

```diff
@@ -273,7 +278,103 @@ should mention how to instantiate those variables/files, etc.
+* achieves the general result its stakeholders desire.
+
+#### Test driven development
+Test-driven development (TDD) is a software development process that relies on the repetition of a very short development cycle: requirements are turned into 
+very specific test cases, then the code is improved so that the tests pass. In simple terms, 
+test cases for each functionality are created and tested first and if the test fails then the new code is written in order to pass the test 
+and making code simple and bug-free.
+
+
+
+Useful links for writing tests in our most common  languages and frameworks: 
+* [Python](https://realpython.com/python-testing/)
+* [React](https://reactjs.org/docs/testing-recipes.html)
+* [Javascript](https://jestjs.io/)
+* [R](https://r-pkgs.org/tests.html)
+
+## Continuous integration and delivery
+
+The CI/CD pipeline is one of the best practices for devops teams to implement, 
+for delivering code changes more frequently and reliably. It is also an agile methodology 
+best practice, as it enables software development teams to focus on meeting business requirements, 
+and code quality because deployment steps are automated.
+
+### Continuous integration
+Developers practicing continuous integration merge their changes back to the main branch as often as possible. 
+The developer's changes are validated by creating a build and running automated tests against the build. 
+By doing so, you avoid the integration hell that usually happens when people wait for release day to merge
+ their changes into the release branch.
+
+Continuous integration puts a great emphasis on testing automation to check that the application is not broken 
+whenever new commits are integrated into the main branch.
+
+#### What you need (cost)
+Your team will need to write automated tests for each new feature, improvement or bug fix.
+You need a continuous integration server that can monitor the main repository and run the tests automatically for every new commits pushed.
+Developers need to merge their changes as often as possible, at least once a day.
+#### What you gain
+Less bugs get shipped to production as regressions are captured early by the automated tests.
+Building the release is easy as all integration issues have been solved early.
+Less context switching as developers are alerted as soon as they break the build and can work on fixing it before they move to another task.
+
+### Continuous delivery
+Continuous delivery is an extension of continuous integration to make sure that you can release new changes to your 
+customers quickly in a sustainable way. This means that on top of having automated your testing, you also have automated
+ your release process and you can deploy your application at any point of time by clicking on a button.
+ 
+#### What you need (cost)
+You need a strong foundation in continuous integration and your test suite needs to cover enough of your codebase.
```

Comment:
Something about how good it is to have something consuming delivered releases in a timely manner to catch issues in a real environment that testing doesn't?

(It feels like I want to link this to agile development as well with customer interaction, but I'm not sure where/how.)